### DISCUSSION TITLE
Lesson 9: Chainlink VRF forever pending (right at the end of the lesson)

### DISCUSSION THREAD

**Discussion by Ynyesto** [Cyfrin/foundry-full-course-cu#1084] [Q&A]
Hi, guys!

I followed along lesson 9 doing everything exactly like Patrick did, except for creating the VRF contract from GUI, which I decided I wanted to see get done when deploying the Raffle contract, so I left the parameter subscriptionId in the NetworkConfig to 0. 

However, even though all of my tests passed (except for those that couldn't without having a predeployed VRF contract and passing its subscriptionId), I get stuck when the automation runs:

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/33223441/36918b19-d135-453b-a100-01e5f5a11062)

As you can see in the screenshot above, the VRF call stays as forever pending. So my question is: has any of you actually got this working WITHOUT a subscriptionId on the NetworkConfig (that is subscriptionId = 0)??

If so, did you encounter any issues? I followed along exactly what Patrick did except for that, so my guess is that it's failing somewhere, although not when creating the VRF contract, which worked fine, as you can also see in the image, since it is from a VRF contract.

Anyway, any insights would be much appreciated. Thanks in advance!

**Comment by 0xusmanf** [comment]
I'm not sure why this is happening but if you can share your repo then maybe I will figure out the problem by recreating it on my end.

**Reply by Ynyesto** [reply]
Hey, man! Thanks very much! Here's the [link ](https://github.com/arynyestos/SolidityFoundryCourse-Lesson9) to the repo!

**Reply by 0xusmanf** [reply]
Hi, thanks for the repo, the problem seems to be with the Chainlink VRF subscription balance. If you fund your subscription with 3 Link tokens then your request for `fulfillRandomWords` will stay stuck, but if you fund it with more than 3 Link tokens then it will work normally.

You can also fund your exciting subscription balance and make it more than 3 Link tokens and your pending request will successfully execute.

**Reply by Ynyesto** [reply]
Hmm... That's odd, why does that happen? 3 LINK is the amount Patrick uses in his video, shouldn't that work??

**Reply by 0xusmanf** [reply]
Yup that's odd and it should work, but I think some recent upgrade to the VRF caused this bug.

**Reply by Ynyesto** [reply]
OK. Now the problem with this is the request already expired, so the contract now is in CALCULATING state, which means no new players can enter the raffle, but also there's no way it will ever finish calculating, so I had to redeploy. This time I funded with 6 LINK and it worked perfectly, so thanks very much for that. How did you know that the issue was insufficient funding?? I didn't see any hints of that!!

**Reply by 0xusmanf** [reply]
Yes you are right there is no hint for that, I spent quite some time experimenting with it, trying different things to make it work. And eventually, when I funded the subscription with more than 3 links it worked.
